### ISSUE TITLE
Libby Downloader hangs

### ISSUE THREAD

**Issue by mjpking** [bookbonobo/libby-download-extension#6]
FireFox 91.11.0esr (64-bit)
Debian 5.10.0-13-amd64

Version 0.3.0

**Comment by mkgtu** [comment]
Same here. The downloader just gets stuck with "line 1" saying "reloading tab"....forever. If I click the "download" icon the tab actually does reload, but still nothing else happens.

Firefox 113.0.1 (64 bit)
Windows 10
![2023-05-15](https://github.com/bookbonobo/libby-download-extension/assets/118212003/6ba75949-b717-404f-9ecd-c1352f385611)

**Comment by sherip99** [comment]
I tried downloading a collection of books (one title) that lacked chapter markers of each book. Libby downloader got each "book" as a track-chapter (ok) but unfortunately the first title was exceptionally long, and the download of that track ended prematurely ie only about half of its audio got downloaded.

**Comment by bookbonobo** [comment]
I've improved the error reporting on this in [0.4.0](https://github.com/bookbonobo/libby-download-extension/releases/tag/0.4.0). What seems to be happening is sometimes an incomplete network response is sent where I'm expecting to see book metadata. I've been able to fix it by reloading the tab while the extension is still running. There should be a new error in the task management table recommending you do that when this situation comes up. I'm going to close this for now, but if you continue to see this behavior feel free to open a new issue (preferably with the debug logs https://github.com/bookbonobo/libby-download-extension/issues/9#issuecomment-1598745150).
